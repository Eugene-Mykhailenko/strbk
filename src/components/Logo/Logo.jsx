import React from 'react';
import PropTypes from 'prop-types';
import styles from './Logo.scss';

const Logo = (props) => {
  const { onClick, link, width, height, imageUrl, alt } = props;
  const logoStyles = { width, height };

  return (
    <a href={link} onClick={onClick} className={styles.logo} style={logoStyles}>
      <img src={imageUrl} alt={alt} />
    </a>
  );
};

Logo.propTypes = {
  /** It's an action when the user clicks on the link. */
  onClick: PropTypes.func.isRequired,
  /** Logo link URL. */
  link: PropTypes.string,
  /** Logo image width. */
  width: PropTypes.number,
  /** Logo image height. */
  height: PropTypes.number,
  /** Logo image URL. */
  imageUrl: PropTypes.string,
  /** Logo image alternative text. */
  alt: PropTypes.string,
};

Logo.defaultProps = {
  link: undefined,
  width: undefined,
  height: undefined,
  imageUrl: undefined,
  alt: '',
};

export default Logo;
