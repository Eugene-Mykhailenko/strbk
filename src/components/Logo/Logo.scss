//logo styles
.logo {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  align-self: center;

  img {
    display: block;
    width: auto;
    height: auto;
    max-width: 100%;
    max-height: 100%;
  }
}
