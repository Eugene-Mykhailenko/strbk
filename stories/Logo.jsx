import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text, number } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import styles from './Stories.scss';
import Logo from '../src/components/Logo/Logo';

const addInfo = withInfo({
  text: `Logo component to mobile or desktop`,
  inline: true,
});

const wrapper = () => (
  <div className={styles.container}>
    <div className={styles.col}>
      <Logo
        link={text('Link href:', '#')}
        width={number('width:', 188)}
        height={number('height:')}
        imageUrl={text(
          'Image URL:',
          'https://media.licdn.com/dms/image/C560BAQFt7d4CSp-Ipg/company-logo_200_200/0?e=2159024400&v=beta&t=G0mATzHeqt9aCqaSS1iDdeOBJkyrmUg7eDZU3ZdJ-Eo',
        )}
        onClick={action('On click handler')}
      />
    </div>
  </div>
);

storiesOf('Logo', module).add('Logo', addInfo(wrapper));
