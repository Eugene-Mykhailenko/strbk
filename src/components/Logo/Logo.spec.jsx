import React from 'react';
import Logo from './Logo';

const config = {
  onClick: jest.fn(),
  link: 'https://air.parimatch.com/',
  imageUrl: 'https://air.parimatch.com/img/logos/pm-logo.svg',
};

describe('Logo', () => {
  beforeEach(() => {
    config.onClick.mockClear();
  });

  it('should be with link', () => {
    const logo = shallow(<Logo link={config.link} onClick={config.onClick} />);

    expect(logo.prop('href')).toEqual(config.link);
    expect(logo).toMatchSnapshot();
  });

  it('should handle click event', () => {
    const button = mount(<Logo onClick={config.onClick} />);

    button.find('.logo').simulate('click');

    expect(config.onClick).toHaveBeenCalledTimes(1);
  });

  it('should be with imageUrl', () => {
    const logo = shallow(
      <Logo imageUrl={config.imageUrl} onClick={config.onClick} />,
    );

    expect(logo.childAt(0).prop('src')).toEqual(config.imageUrl);
    expect(logo).toMatchSnapshot();
  });
});
