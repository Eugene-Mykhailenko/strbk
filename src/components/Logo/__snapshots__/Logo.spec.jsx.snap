// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should be with imageUrl 1`] = `
<a
  className="logo"
  onClick={[MockFunction]}
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <img
    alt=""
    src="https://air.parimatch.com/img/logos/pm-logo.svg"
  />
</a>
`;

exports[`Logo should be with link 1`] = `
<a
  className="logo"
  href="https://air.parimatch.com/"
  onClick={[MockFunction]}
  style={
    Object {
      "height": undefined,
      "width": undefined,
    }
  }
>
  <img
    alt=""
  />
</a>
`;
